### PR TITLE
Inverse Zoomd and RandZoomd

### PR DESCRIPTION
**Pull request by rijobro**
Inverse `Zoomd` and `RandZoomd`

#1515

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/spatial/dictionary.py [hunk 1]
```diff
@@ -1311,6 +1338,29 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
...
         return d
 
+    def inverse(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
+        d = deepcopy(dict(data))
+        for key, mode, padding_mode, align_corners in self.key_iterator(
+            d, self.mode, self.padding_mode, self.align_corners
+        ):
+            transform = self.get_most_recent_transform(d, key)
+            # Create inverse transform
+            zoom = np.array(transform[InverseKeys.EXTRA_INFO.value]["zoom"])
```
Comment (wyli) [inline monai/transforms/spatial/dictionary.py:1348]: I think `_do_transform` flag should be checked somewhere in this method?

Comment (rijobro) [reply monai/transforms/spatial/dictionary.py]: thanks, updated

**Review by wyli — APPROVED** [review]
thanks!

File: monai/transforms/spatial/dictionary.py [hunk 2]
```diff
@@ -1338,6 +1362,8 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
         self.randomize()
         d = dict(data)
         if not self._do_transform:
+            for key in self.keys:
+                self.push_transform(d, key, extra_info={"zoom": self._zoom})
```
Comment (wyli) [inline monai/transforms/spatial/dictionary.py:1366]: perhaps extra_info `self._zoom` could have different sizes (on/off _do_transform) in a batch loader, then they are not compatible with the collate https://github.com/Project-MONAI/MONAI/issues/1798

Comment (rijobro) [reply monai/transforms/spatial/dictionary.py:1366]: Yeah, you're right. You can see there's a few `ensure_tuple(self._zoom)` that come a few lines later, so they just need to be moved ahead.

**Pull request merged**